passkey-types/src/webauthn/attestation.rs

```diff
@@ -665,4 +930,207 @@ mod tests {
         assert_eq!(deserialized.public_key.timeout, Some(300_000));
         assert_eq!(deserialized.public_key.pub_key_cred_params.len(), 2)
     }
+
+    #[test]
+    fn test_client_data_type_to_string() {
+        let payment_get = ClientDataType::PaymentGet;
+        assert_eq!(payment_get.to_string(), "payment.get");
+
+        let webauthn_get = ClientDataType::Get;
+        assert_eq!(webauthn_get.to_string(), "webauthn.get");
+    }
+
+    #[test]
+    fn test_client_data_serialization() {
+        // This is the raw client data json byte buffer returned by a webauthn assertion
+        let expected_client_data_bytes: [u8; 134] = [
+            123, 34, 116, 121, 112, 101, 34, 58, 34, 119, 101, 98, 97, 117, 116, 104, 110, 46, 103,
+            101, 116, 34, 44, 34, 99, 104, 97, 108, 108, 101, 110, 103, 101, 34, 58, 34, 90, 69,
+            118, 77, 102, 108, 90, 68, 99, 119, 81, 74, 109, 97, 114, 73, 110, 110, 89, 105, 56,
+            56, 112, 120, 45, 54, 72, 90, 99, 118, 50, 85, 111, 120, 119, 55, 45, 95, 74, 79, 79,
+            84, 103, 34, 44, 34, 111, 114, 105, 103, 105, 110, 34, 58, 34, 104, 116, 116, 112, 58,
+            47, 47, 108, 111, 99, 97, 108, 104, 111, 115, 116, 58, 52, 48, 48, 48, 34, 44, 34, 99,
+            114, 111, 115, 115, 79, 114, 105, 103, 105, 110, 34, 58, 102, 97, 108, 115, 101, 125,
+        ];
```

Comment:
**Issue:** Maintaining these binary blobs is not a great experience. These should follow the JSON string recommendation mentioned [here](https://github.com/1Password/passkey-rs/pull/6#discussion_r1406347985). This applies to the two below in the next test as well

Reply:
Ok will remove

Reply:
In the above case I stored the challenge as an array of bytes because that's what it's type is. Granted I could have put it as base64Url instead, but generally there is no internal structure to those arrays.

For the tests, please see the example I provided in the comment [here](https://github.com/1Password/passkey-rs/pull/6#discussion_r1406347985). The hard coded json _does not need to be formatted_. I agree with you that we need to test the byte serialization (lack of white space for example). however that does not preclude using the string representation. Taking this blob through cyberchef for example:

```rust
let expected_client_data_bytes: &[u8] = r#"{"type":"webauthn.get","challenge":"ZEvMflZDcwQJmarInnYi88px-6HZcv2Uoxw7-_JOOTg","origin":"http://localhost:4000","crossOrigin":false}"#.as_bytes();
```

Also it does not matter that this line will be long, I just need to be able to change it easily if necessary. A byte array is not as easy to maintain.

Reply:
@Progdrasil ok, i've done as you asked and converted it into a string, lmk if it looks good to you